src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -595,6 +595,29 @@ internal bool IsEntryPointCandidate
             get { return IsStatic && Name == WellKnownMemberNames.EntryPointMethodName; }
         }
 
+        internal bool ReturnsTaskOrTaskOfInt(CSharpCompilation compilation)
```

Comment:
>internal bool ReturnsTaskOrTaskOfInt(CSharpCompilation compilation) [](start = 8, length = 67)

According to the C# language specification type of the task's result is determined based on the return type of the GetResult method, rather than based on the shape of the task type itself. I think it would make sense to follow the same approach for figuring out if task-returning Main "returns" a value and what is the type of that value.